jerry-core/ecma/base/ecma-globals.h

```diff
@@ -614,11 +554,55 @@ typedef struct
 #define ECMA_PROPERTY_IS_PROPERTY_PAIR(property_header_p) \
   ((property_header_p)->types[0] != ECMA_PROPERTY_TYPE_HASHMAP)
 
+/**
+ * Property value of all internal properties
+ */
+#define ECMA_PROPERTY_INTERNAL \
+  (ECMA_PROPERTY_FLAG_DATA | (ECMA_DIRECT_STRING_SPECIAL << ECMA_PROPERTY_NAME_TYPE_SHIFT))
+
+/**
+ * Checks whether a property is internal property
+ */
+#define ECMA_PROPERTY_IS_INTERNAL(property) ((property) >= ECMA_PROPERTY_INTERNAL)
+
+/**
+ * Checks whether a property is raw data or accessor property
+ */
+#define ECMA_PROPERTY_IS_RAW(property) \
```

Comment:
I like something like (property & 0b00011000)
Without the need of compare operator
Any definition of property bits? 
property is 8 bits long, what's the meaning of each bits?

Reply:
Good bit magic is essential for high performance. This condition checks that the highest two bits are not set *without* using a temporary register. The alternative ´(x & 0xc0) != 0xc0´ requires two operations and a temporary register.
Property bits are described in ecma_property_flags_t.